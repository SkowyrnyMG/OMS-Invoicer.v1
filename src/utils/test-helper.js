import React from 'react';
import { Router } from 'react-router-dom';
import { render, fireEvent } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { createMemoryHistory } from 'history';
import { theme } from 'themes/theme';

export const TestThemeProvider = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

const renderWithRouter = (ui, { route = '/', ...renderOptions } = {}) => {
  const history = createMemoryHistory({ initialEntries: [route] });
  const utils = render(<Router history={history}>{ui}</Router>, renderOptions);
  return {
    ...utils,
    history,
  };
};

const leftClick = { button: 0 };

export { leftClick, renderWithRouter, fireEvent };
