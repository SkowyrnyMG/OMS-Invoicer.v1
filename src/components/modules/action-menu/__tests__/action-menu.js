import React from 'react';

import { renderWithReduxRouter, screen } from 'utils/tests/test-helper';
import ActionMenu from '../action-menu';

describe('ActionMenu', () => {
  test('should display in the document with correct styles', () => {
    renderWithReduxRouter(
      <ActionMenu>
        <button>Test button</button>
      </ActionMenu>,
    );

    const ActionMenuComponent = screen.getByRole('complementary', {
      name: '',
    });

    expect(ActionMenuComponent).toBeInTheDocument();
    expect(ActionMenuComponent).toMatchSnapshot();
  });
});
