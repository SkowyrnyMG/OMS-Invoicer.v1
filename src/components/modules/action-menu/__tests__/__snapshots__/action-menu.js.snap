// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionMenu should display in the document with correct styles 1`] = `
.c0 > * {
  margin-bottom: 0.5rem;
}

@media (max-width:767px) {
  .c0 {
    grid-row: 1 / 2;
    display: grid;
    grid-template-columns: repeat(3,1fr);
    grid-column-gap: 2%;
    justify-items: center;
    margin-bottom: 2rem;
  }

  .c0 > * {
    margin-bottom: 0.5rem;
    margin-right: 0.5rem;
  }
}

@media (max-width:570px) {
  .c0 {
    grid-template-columns: repeat(2,1fr);
  }
}

@media (max-width:374px) {
  .c0 {
    grid-template-columns: 1fr;
  }
}

<aside
  class="c0"
>
  <button>
    Test button
  </button>
</aside>
`;
